L1Trigger/L1THGCal/src/backend/HGCalMulticlusteringHistoImpl.cc

```diff
@@ -215,16 +220,84 @@ std::vector<GlobalPoint> HGCalMulticlusteringHistoImpl::computeMaxSeeds( const H
                 isMax &= MIPT_seed>MIPT_SW;
                 isMax &= MIPT_seed>MIPT_NW;
 
+                if(isMax){
+                    float ROverZ_seed = kROverZMin_ + (bin_R+0.5) * (kROverZMax_-kROverZMin_)/nBinsRHisto_;
+                    float phi_seed = -M_PI + (bin_phi+0.5) * 2*M_PI/nBinsPhiHisto_;
+                    float x_seed = ROverZ_seed*cos(phi_seed);
+                    float y_seed = ROverZ_seed*sin(phi_seed);
+
+                    seedPositions.emplace_back( std::make_pair( GlobalPoint(x_seed,y_seed,z_side), MIPT_seed) );
+                    primarySeedPositions[bin_R][bin_phi][z_side] =  true;
 
 
-                if(isMax){
+                    vetoPositions[bin_R][binLeft][z_side] = true;
+                    vetoPositions[bin_R][binRight][z_side] = true;
+                    if ( bin_R>0 ) {
+                      vetoPositions[bin_R-1][bin_phi][z_side] = true;
+                      vetoPositions[bin_R-1][binRight][z_side] = true;
+                      vetoPositions[bin_R-1][binLeft][z_side] = true;
+                    }
+                    if ( bin_R<(int(nBinsRHisto_)-1) ) {
+                      vetoPositions[bin_R+1][bin_phi][z_side] = true;
+                      vetoPositions[bin_R+1][binRight][z_side] = true;
+                      vetoPositions[bin_R+1][binLeft][z_side] = true;
+                    }
 
+                }
+
+            }
 
+        }
+
+    }
+
+
+    //Search for secondary seeds
+
+    for(int z_side : {-1,1}){
+
+        for(int bin_R = 0; bin_R<int(nBinsRHisto_); bin_R++){
+
+            for(int bin_phi = 0; bin_phi<int(nBinsPhiHisto_); bin_phi++){
+
+              //Cannot be a secondary seed if already a primary seed, or adjacent to primary seed
+              if ( primarySeedPositions[bin_R][bin_phi][z_side] || vetoPositions[bin_R][bin_phi][z_side] ) continue;
+
+                float MIPT_seed = histoClusters.at({{z_side,bin_R,bin_phi}});
+                bool isMax = MIPT_seed > histoThreshold_;
```

Comment:
It's not clear what the difference is between a secondary and a primary seed is since both must pass the same threshold and be a local maxima so it looks like that all potential secondaries are already primaries.

Reply:
The important lines are those such as:
```                if (  !vetoPositions[std::make_tuple(bin_R+1,bin_phi,z_side)]  ) isMax &= MIPT_seed>=MIPT_S;```

A seed may be a secondary, even if it is not a local maximum because veto bins (i.e. those adjacent to the primaries) are ignored.

It is the case that all primaries would also be a secondary, therefore we have the line:

```if ( primarySeedPositions[std::make_tuple(bin_R,bin_phi,z_side)] || vetoPositions[std::make_tuple(bin_R,bin_phi,z_side)] ) continue;```

to avoid double counting.

Reply:
Ok thanks. I missed that point.